src/types/blob.rs

```diff
@@ -115,6 +115,12 @@ impl AsRef<[u8]> for BlobRef {
     }
 }
 
+impl AsRef<BlobRef> for [u8] {
```

Comment:
I think it's better not to implement `AsRef` for very commonly-used types like `[u8]`. `to_byte_slice` and `from_byte_slice` all should be part of `BlobEncode`.

Reply:
Yes. Btw, this impl seems useless. I forget why I added it 😅